src/backend/distributed/planner/merge_planner.c

```diff
@@ -857,8 +857,11 @@ ConvertRelationRTEIntoSubquery(Query *mergeQuery, RangeTblEntry *sourceRte,
 	/* set the FROM expression to the subquery */
 	newRangeTableRef->rtindex = SINGLE_RTE_INDEX;
 	sourceResultsQuery->jointree = makeFromExpr(list_make1(newRangeTableRef), NULL);
+
```

Comment:
` requiredAttributes = RequiredAttrNumbersForRelationInternal(mergeQuery, relationRestriction->index);
` 
Here were we getting 1, 3?